apiserver/plane/app/views/workspace/preference.py

```
@@ -0,0 +1,43 @@
+    @allow_permission([ROLE.ADMIN, ROLE.MEMBER, ROLE.GUEST], level="WORKSPACE")
+    def get(self, request, slug):
+        workspace = Workspace.objects.get(slug=slug, owner=request.user)
+    
+        quick_links = WorkspaceHomePreference.objects.filter(user=request.user, workspace_id=workspace.id, key=WorkspaceHomePreference.HomeWidgetKeys.QUICK_LINKS).exists()
+        recents = WorkspaceHomePreference.objects.filter(user=request.user, workspace_id=workspace.id, key=WorkspaceHomePreference.HomeWidgetKeys.RECENTS).exists()
+        my_stickies = WorkspaceHomePreference.objects.filter(user=request.user, workspace_id=workspace.id, key=WorkspaceHomePreference.HomeWidgetKeys.MY_STICKIES).exists()
+        new_at_plane = WorkspaceHomePreference.objects.filter(user=request.user, workspace_id=workspace.id, key=WorkspaceHomePreference.HomeWidgetKeys.NEW_AT_PLANE).exists()
+        quick_tutorial = WorkspaceHomePreference.objects.filter(user=request.user, workspace_id=workspace.id, key=WorkspaceHomePreference.HomeWidgetKeys.QUICK_TUTORIAL).exists()
+
+        if quick_links and recents and my_stickies and new_at_plane and quick_tutorial:
+            return Response(WorkspaceHomePreference.objects.filter(user=request.user, workspace_id=workspace.id).values("user", "workspace", "sort_order", "is_enabled", "key"), status=status.HTTP_201_CREATED)  
```

Ruff (0.8.2)

Line too long (210 > 88)

(E501)